portal/portal-impl/impl/src/java/org/sakaiproject/portal/entityprovider/PortalEntityProvider.java

```diff
@@ -70,4 +106,155 @@ public PortalNotifications handleNotify(EntityView view) {
 		return noti;
 	}
 
+	@EntityCustomAction(action = "socialAlerts", viewKey = EntityView.VIEW_LIST)
+	public ActionReturn getSocialAlerts(EntityView view) {
+
+		String currentUserId = developerHelperService.getCurrentUserId();
+
+		if (currentUserId == null) {
+			throw new SecurityException("You must be logged in to use this service");
+		}
+
+		List<BullhornAlert> alerts = portalService.getSocialAlerts(currentUserId);
+
+		ResourceLoader rl = new ResourceLoader("bullhorns");
+
+		if (alerts.size() > 0) {
+			Map<String, Object> data = new HashMap<String, Object>();
+			data.put("alerts", alerts);
+			data.put("i18n", rl);
+
+			return new ActionReturn(data);
+		} else {
+			Map<String, String> i18n = new HashMap<String, String>();
+			i18n.put("noAlerts", rl.getString("noAlerts"));
+
+			Map<String, Object> data = new HashMap<String, Object>();
+			data.put("message", "NO_ALERTS");
+			data.put("i18n", i18n);
+
+			return new ActionReturn(data);
+		}
+	}
+
+	@EntityCustomAction(action = "clearSocialAlert", viewKey = EntityView.VIEW_LIST)
+	public boolean clearSocialAlert(Map<String, Object> params) {
+
+		String currentUserId = developerHelperService.getCurrentUserId();
+
+		if (currentUserId == null) {
+			throw new SecurityException("You must be logged in to use this service");
+		}
+
+		try {
+			long alertId = Long.parseLong((String) params.get("id"));
+			return portalService.clearSocialAlert(currentUserId, alertId);
+		} catch (Exception e) {
+			log.error("Failed to clear social alert", e);
+		}
+
+		return false;
+	}
+
+	@EntityCustomAction(action = "clearAllSocialAlerts", viewKey = EntityView.VIEW_LIST)
+	public boolean clearAllSocialAlerts(Map<String, Object> params) {
+
+		String currentUserId = developerHelperService.getCurrentUserId();
+
+		if (currentUserId == null) {
+			throw new SecurityException("You must be logged in to use this service");
+		}
```

Comment:
Seems like you do this in every method, might be a good candidate for extracting to convenience method to reduce code duplication.